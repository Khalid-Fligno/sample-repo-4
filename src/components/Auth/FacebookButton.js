import React from 'react';
import { Dimensions, TouchableOpacity, Text, StyleSheet, Image } from 'react-native';
import PropTypes from 'prop-types';
import colors from '../../styles/colors';
import fonts from '../../styles/fonts';

const { width } = Dimensions.get('window');

const FacebookButton = ({
  title,
  onPress,
}) => (
  <TouchableOpacity
    onPress={onPress}
    style={styles.button}
  >
    <Image
      source={require('../../../assets/icons/facebook-icon-white.png')}
      style={styles.facebookIcon}
    />
    <Text
      style={styles.buttonText}
    >
      {title}
    </Text>
  </TouchableOpacity>
);

FacebookButton.propTypes = {
  title: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};

const styles = StyleSheet.create({
  button: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'rgb(59,89,152)',
    height: 50,
    width: width - 30,
    marginTop: 8,
    borderRadius: 4,
    shadowColor: colors.charcoal.dark,
    shadowOpacity: 0.5,
    shadowOffset: { width: 0, height: 2 },
    shadowRadius: 3,
  },
  facebookIcon: {
    width: 32,
    height: 32,
    marginRight: 10,
  },
  buttonText: {
    marginTop: 4,
    fontFamily: fonts.bold,
    fontSize: 15,
    color: colors.white,
  },
});

export default FacebookButton;
