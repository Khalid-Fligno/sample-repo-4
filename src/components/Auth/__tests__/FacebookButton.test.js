import React from 'react';
import renderer from 'react-test-renderer';
import FacebookButton from '../FacebookButton';

describe('FacebookButton', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<FacebookButton title="Test" onPress={jest.fn()} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
