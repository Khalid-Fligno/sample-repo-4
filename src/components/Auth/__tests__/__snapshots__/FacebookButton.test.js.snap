// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacebookButton renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(59,89,152)",
      "borderRadius": 4,
      "flexDirection": "row",
      "height": 45,
      "justifyContent": "center",
      "marginTop": 8,
      "opacity": 1,
      "shadowColor": "rgb(35, 36, 44)",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 3,
      "width": 720,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 32,
        "marginRight": 10,
        "width": 32,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "rgb(255, 255, 255)",
        "fontFamily": "GothamBold",
        "fontSize": 15,
        "marginTop": 4,
      }
    }
  >
    Test
  </Text>
</View>
`;
